server/container/guice/cassandra-guice/src/main/java/org/apache/james/modules/mailbox/ElasticSearchStartUpCheck.java

```diff
@@ -0,0 +1,78 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.modules.mailbox;
+
+import javax.inject.Inject;
+
+import org.apache.james.StartUpChecksPerformer;
+import org.apache.james.metrics.es.ESReporterConfiguration;
+import org.apache.james.modules.mailbox.ElasticSearchRestAPI.About;
+import org.apache.james.util.Host;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import feign.RetryableException;
+
+public class ElasticSearchStartUpCheck implements StartUpChecksPerformer.StartUpCheck {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ElasticSearchStartUpCheck.class);
+    private static final String CHECK_NAME = "ElasticSearchStartUpCheck";
+    private static final String CAN_NOT_FIND_VERSION_MESSAGE = "cannot find ElasticSearch version";
+
+    private final Host esHttpHost;
+
+    @Inject
+    public ElasticSearchStartUpCheck(ESReporterConfiguration esConfiguration) {
```

Comment:
Are you specifically checking the ElasticSearch instance used for metric collection?

We should concentrate on the one providing the search feature - they could be different...

Reply:
ESReporterConfiguration is the only instance carrying the ES http host, I want to use it temporally to have a tests suite for this class. Then I will think about the alternative instance